### PR TITLE
[UWP] Use custom button to show OpenUrl actions as links

### LINKED ISSUE #3562: [UWP] Action.OpenUrl buttons should appear as links to accessibility
Per a partner team's accessibility review, `Action.OpenUrl` buttons in UWP should be reported as a `hyperlink` type so their behavior lines up with their name.

### PR DESCRIPTION
**Pull request by paulcam206**
Fixes #3562

## Description
OpenUrl actions should appear as a link to accessibility. Unfortunately, to make that work, we need to create a new type of button. Unfortunately, to do that in a reasonable way, we need to use C++/CX or C++/WinRT rather than going directly against the ABI. C++/CX is a non-starter for us, but C++/WinRT is just right. 

## How Verified
* [Accessibility Insights](https://accessibilityinsights.io/)
* UWP visualizer
* UWP TestApp

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3555)

### CODE REVIEW

**Comment by shalinijoshi19** [comment]
@almedina-ms , what's up with the .net failures? Did we ever recover from the .NET core failures from before? 

<snip>
hase 1: "dotnet : Testhost process exited with error: It was not possible to find any compatible framework version At E:\BA\1\s\source\dotnet\RunAllTests.ps1:14 char:1 + dotnet test .\test\AdaptiveCards.Test + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ + CategoryInfo : NotSpecified: (Testhost proces...amework version:String) [], RemoteException + FullyQualifiedErrorId : NativeCommandError The specified framework 'Microsoft.NETCore.App', version '2.0.0' was not found. - The following frameworks were found: 3.0.0 at [E:\BA\_tool\dotnet\ shared\Microsoft.NETCore.App] You can resolve the problem by installing the specified framework and/or SDK. The .NET Core frameworks can be found at: - https://aka.ms/dotnet-download . Please check the diagnostic logs for more information. "
</snip>

File: source/uwp/Renderer/lib/pch.h [hunk 1]
```diff
@@ -57,3 +57,9 @@
 #include <windows.foundation.collections.h>
 #include <windows.ui.xaml.shapes.h>
 #include <windows.ui.xaml.markup.h>
+
+#include <winrt/base.h>
+#include <winrt/Windows.UI.Xaml.Automation.h>
+#include <winrt/Windows.UI.Xaml.Automation.Peers.h>
+#include <winrt/Windows.UI.Xaml.Controls.h>
```
Comment (shalinijoshi19) [inline source/uwp/Renderer/lib/pch.h]: nice! Thanks for getting this done and getting us started on the winrt/cpp route !

**Comment by almedina-ms** [comment]
> 
> 
> @almedina-ms , what's up with the .net failures? Did we ever recover from the .NET core failures from before?
> hase 1: "dotnet : Testhost process exited with error: It was not possible to find any compatible framework version At E:\BA\1\s\source\dotnet\RunAllTests.ps1:14 char:1 + dotnet test .\test\AdaptiveCards.Test + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ + CategoryInfo : NotSpecified: (Testhost proces...amework version:String) [], RemoteException + FullyQualifiedErrorId : NativeCommandError The specified framework 'Microsoft.NETCore.App', version '2.0.0' was not found. - The following frameworks were found: 3.0.0 at [E:\BA\_tool\dotnet\ shared\Microsoft.NETCore.App] You can resolve the problem by installing the specified framework and/or SDK. The .NET Core frameworks can be found at: - https://aka.ms/dotnet-download . Please check the diagnostic logs for more information. "

@shalinijoshi19 The build breaks from before had already been solved but it looks like the fix didn't work correctly, I'll move the .NET Core 3.0 to another solution

File: source/uwp/Renderer/lib/ActionHelpers.cpp [hunk 2]
```diff
@@ -274,10 +276,25 @@ namespace AdaptiveNamespace::ActionHelpers
...
+        ABI::AdaptiveNamespace::ActionType actionType;
+        RETURN_IF_FAILED(action->get_ActionType(&actionType));
+
+        // now construct an appropriate button for the action type
+        ComPtr<IButton> button;
+        if (actionType == ABI::AdaptiveNamespace::ActionType_OpenUrl)
+        {
+            // OpenUrl buttons should appear as links for accessibility purposes, so we use our custom LinkButton.
+            auto linkButton = winrt::make<LinkButton>();
+            button = linkButton.as<IButton>().detach();
```
Comment (shalinijoshi19) [inline source/uwp/Renderer/lib/ActionHelpers.cpp]: > button = linkButton.as<IButton>().detach [](start = 11, length = 41)

Can we just add some very basic unit tests to test construction etc of the new LinkButton object type? Also how much of manual testing should we try for this? ANy backcompat issues that you anticipate at all ?

**Comment by shalinijoshi19** [comment]
@paulcam206 is there a github issue tracking this at all?

**Comment by paulcam206** [comment]
> @paulcam206 is there a github issue tracking this at all?

I just filed #3562 to track this request

**Comment by paulcam206** [comment]
@nesalang / @almedina-ms: need one more review :)

**Comment by shalinijoshi19** [comment]
> @nesalang / @almedina-ms: need one more review :)



> @nesalang / @almedina-ms: need one more review :)

Also added Jevan/Mike from XAML we'd been working with offline for a quick "deko"
